client/galaxy/scripts/entry/panels/tool-panel.js

```diff
@@ -52,9 +52,21 @@ const ToolPanel = Backbone.View.extend({
                 title: _l("Show favorites"),
                 icon: "fa fa-star-o",
                 onclick: e => {
-                    $("#tool-search-query")
-                        .val("#favorites")
-                        .trigger("change");
+                    $(".panel-header-button .fa").toggleClass("fa-star-o fa-star");
+                    //hide the tooltip
+                    $(".panel-header-button").tooltip('hide');
+                    if($("#tool-search-query").val().indexOf("#favorites")!= -1)
+                        {
+                            $("#tool-search-query").val("");
```

Comment:
Can you try to assign `$("#tool-search-query")` to a variable and then use it at different places? Similarly for other dom elements if possible

Reply:
Sure. Will do it!